server/asset/eth/coiner.go

```diff
@@ -117,7 +117,7 @@ func (backend *Backend) newSwapCoin(coinID []byte, contract []byte, sct swapCoin
 			return nil, fmt.Errorf("tx %v does not contain initiation with secret hash %x", txHash, secretHash)
 		}
 		counterParty = &initiation.Participant
-		secretHash = initiation.SecretHash
+		secretHash = initiation.SecretHash // no-op
```

Comment:
Actually, I think it would be right to compare the secretHash from the contract data against the one from the call data here.  That's why I removed `validateRedeem`.